bin/public/installESpower.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python
+'''
+Install DHCP Server
+
+This dhcp server is mainly used to get KVM guest installation to work with
+kickstarts.
+
+http://www.yolinux.com/TUTORIALS/DHCP-Server.html
+http://www.linuxhomenetworking.com/wiki/index.php/Quick_HOWTO_:_Ch08_:_Configuring_the_DHCP_Server
+http://www.howtoforge.com/dhcp_server_linux_debian_sarge
+
+'''
+
+__author__ = "mattias.hemmingsson@fareoffice.com"
+__copyright__ = "Copyright 2014, The System Console project"
+__maintainer__ = "Mattias Hemmingsson"
+__email__ = "syco@cybercow.se"
+__credits__ = ["Daniel Lindh"]
+__license__ = "???"
+__version__ = "1.0.0"
+__status__ = "Production"
+
+import general
+from general import x
+from scopen import scOpen
+import app
+import version
+import os
+import iptables
+
+# The version of this module, used to prevent the same script version to be
+# executed more then once on the same host.
+SCRIPT_VERSION = 1
+CONF_SOURCE='/opt/syco/usr/syco-private/var/'
+
+def build_commands(commands):
+    commands.add("install-espower", install_espower, help="Install power modules for elastcisearch")
+    commands.add("uninstall-espower", uninstall_espower, help="Uninstall the power modules for elastic search")
+
+
+def install_espower(args):
+	'''
+	Installation of Elastic search passing rule
+	
+	'''
+	install_rabbit()
+	install_logstash()
+	config_rabbitmq()
+	config_logstash()
+	# Adding iptables rules
+	iptables.add_rabbitmq_chain()
+	iptables.save()
+	
+
+	print("Go to http://ip-address:15672 for rabbit mq ")
+
+def uninstall_espower(args):
+	x('yum remove rabbitmq-server -y')
+	#x('yum remove erlang -y')
+	x('rm -rf /opt/logstash')
+	x('rm -rf /etc/logstash')
+	x('rm -rf /etc/rabbitmq')
+	x('rm -rf /etc/init.d/shipper')
+	x('rm -rf /etc/init.d/index') 
+
+
+
+
+def install_rabbit():
+	'''
+	Installa and setup the rabbit mq server.
+	'''
+
+
+	#check if epel is install else installed
+	if not os.path.isfile("yum install rabbitmq-server -y/etc/yum.repos.d/epel.repo"):
+		x('yum localinstall http://download.fedoraproject.org/pub/epel/6/i386/epel-release-6-8.noarch.rpm -y')
+	x('yum install erlang -y')
+	x('rpm --import http://www.rabbitmq.com/rabbitmq-signing-key-public.asc')
+	x('yum install rabbitmq-server -y')
+
+
+
+def install_logstash():
+    '''
+    Download and install logstash
+    '''
+    x("curl -O  https://download.elasticsearch.org/logstash/logstash/logstash-1.4.2.tar.gz")
+    x('mv logstash-1.4.2.tar.gz /opt/logstash.tar.gz')
+    x('tar -zxvf /opt/logstash.tar.gz -C /opt/')
+    x('rm /opt/logstash.tar.gz')
+    x('rm -rf /opt/logstash')
+    x('ln -s /opt/logstash-1* /opt/logstash')
+
+
+
+def config_logstash():
+	'''
+	There are now defult config for logstash
+	Copy config from
+	1. First from syco-private
```

Comment:
added to get first /var config to use. And then work from that /var on to others service